layer/clients/data_catalog.py

```diff
@@ -439,3 +427,34 @@ def _list_datasets(self, sort_fields: Sequence[SortField] = ()) -> List[PBDatase
                 ),
             ).datasets
         )
+
+
+def _get_batch_chunks(
+    batch: pyarrow.RecordBatch, max_chunk_size_bytes: int = 4_000_000
+) -> Generator[pyarrow.RecordBatch, None, None]:
+    """
+    Slice the batch into the chunks, based on average row size,
+    but not exceeding the maximum chunk size.
+    """
+
+    bytes_per_row = batch.nbytes / batch.num_rows  # average row size
+    max_num_rows = int(max_chunk_size_bytes / bytes_per_row)  # rows per chunk
+    if max_num_rows == 0:
```

Comment:
It would be easier to read if we compare `if bytes_per_row > max_chunk_size_bytes`  instead of the result of the integer division.